examples/tv-casting-app/tv-casting-common/core/CastingPlayer.h

```diff
@@ -119,22 +119,39 @@ class CastingPlayer : public std::enable_shared_from_this<CastingPlayer>
+     * this will execute the User Directed Commissioning (UDC) process.
      *
-     * @param onCompleted for success - called back with CHIP_NO_ERROR and CastingPlayer *.
-     * For failure - called back with an error and nullptr.
+     * @param connectionCallbacks contains the ConnectCallback and CommissionerDeclarationCallback (Optional).
      * @param commissioningWindowTimeoutSec (Optional) time (in sec) to keep the commissioning window open, if commissioning is
      * required. Needs to be >= kCommissioningWindowTimeoutSec.
      * @param idOptions (Optional) Parameters in the IdentificationDeclaration message sent by the Commissionee to the Commissioner.
      * These parameters specify the information relating to the requested commissioning session.
      * Furthermore, attributes (such as VendorId) describe the TargetApp that the client wants to interact with after commissioning.
-     * If this value is passed in, VerifyOrEstablishConnection() will force User Directed Commissioning, in case the desired
+     * If this value is passed in, VerifyOrEstablishConnection() will force UDC, in case the desired
      * TargetApp is not found in the on-device CastingStore.
      */
-    void VerifyOrEstablishConnection(ConnectCallback onCompleted,
+    void VerifyOrEstablishConnection(ConnectionCallbacks connectionCallbacks,
                                      unsigned long long int commissioningWindowTimeoutSec = kCommissioningWindowTimeoutSec,
                                      IdentificationDeclarationOptions idOptions           = IdentificationDeclarationOptions());
 
+    /**
+     * @brief Continues the UDC process during the Commissioner-Generated passcode commissioning flow by sending a second
+     * IdentificationDeclaration to Commissioner containing CommissionerPasscode and CommissionerPasscodeReady set to true. At this
+     * point it is assumed that the following have occurred:
+     * 1. Client has handled the Commissioner's CommissionerDecelration message with PasscodeDialogDisplayed and
+     * CommissionerPasscode set to true.
+     * 2. Client prompted user to input Passcode from Commissioner.
+     * 3. Client has updated the commissioning session's PAKE verifier using the user input Passcode by updating the CastingApps
+     * CommissionableDataProvider
+     * (matter::casting::core::CastingApp::GetInstance()->UpdateCommissionableDataProvider(CommissionableDataProvider)).
+     *
+     * @param connectionCallbacks contains the ConnectCallback and CommissionerDeclarationCallback (Optional).
+     * @param commissioningWindowTimeoutSec (Optional) time (in sec) to keep the commissioning window open, if commissioning is
+     * required. Needs to be >= kCommissioningWindowTimeoutSec.
+     */
+    void ContinueConnecting(ConnectionCallbacks connectionCallbacks,
+                            unsigned long long int commissioningWindowTimeoutSec = kCommissioningWindowTimeoutSec);
```

Comment:
Why do we need to pass commissioningWindowTimeoutSec in ContinueConnecting() at all, it was already passed in to VerifyOrEstablishConnection. Similarly connectionCallbacks should not be a parameter here either.

Reply:
For flexibility in their implementation, the client might want to specify a diferent timeout or callbacks for ContinueConnecting() depending on how they implement their UX. They might have transitioned to a different page in the UX so using a  new callback might be easier to use than the old one. In Java for example callbacks can be defined in the in the function call.